includes/REST/DummyDataController.php

```diff
@@ -0,0 +1,950 @@
+                    'permission_callback' => array( $this, 'get_permissions_check' ),
+                ),
+            )
+        );
+    }
+
+    /**
+     * Returns dokan import status.
+     *
+     * @since DOKAN_SINCE
+     *
+     * @return array
+     */
+    public function import_dummy_data_status() {
+        $status = [
+            'import_status' => get_option( 'dokan_dummy_data_import_success', 'no' ),
+        ];
+
+        return rest_ensure_response( $status );
+    }
+
+        /**
+     * Imports dummy vendors and products.
+     *
+     * @since DOKAN_SINCE
+     *
+     * @return array
+     */
+    public function import_dummy_data( $request ) {
+        $_post_data = $request->get_params();
+
+        $vendor_products = isset( $_post_data['vendor_products'] ) ? wp_unslash( wc_clean( $_post_data['vendor_products'] ) ) : [];
+        $vendor_data     = isset( $_post_data['vendor_data'] ) ? wp_unslash( wc_clean( $_post_data['vendor_data'] ) ) : [];
+        $vendor_index    = isset( $_post_data['vendor_index'] ) ? absint( $_post_data['vendor_index'] ) : 0;
+        $total_vendors   = isset( $_post_data['total_vendors'] ) ? absint( $_post_data['total_vendors'] ) : 0;
+
+        $created_vendor          = dokan()->dummy_data_importer->create_dummy_vendor( $vendor_data );
+        $created_products_result = dokan()->dummy_data_importer->create_dummy_products_for_vendor( $created_vendor->id, $vendor_products );
+
+        if ( $vendor_index + 1 >= $total_vendors ) {
+            update_option( 'dokan_dummy_data_import_success', 'yes', true );
+        }
+
+        $imported = [
+            'result'       => $created_products_result,
+            'vendor_index' => $vendor_index + 1,
+        ];
+
+        return rest_ensure_response( $imported );
+    }
+
+    /**
+     * Clears dokan dummy data.
+     *
+     * @since DOKAN_SINCE
+     *
+     * @return string
+     */
+    public function clear_dummy_data() {
+        $result = dokan()->dummy_data_importer->clear_all_dummy_data();
+        delete_option( 'dokan_dummy_data_import_success' );
+
+        $message = [
+            'message' => $result,
+        ];
+
+        return rest_ensure_response( $message );
+    }
+
+    /**
+     * Checking if have any permission.
+     *
+     * @since DOKAN_SINCE
+     *
+     * @return boolean
+     */
+    public function get_permissions_check() {
+        $permission = true;
+
+        if ( empty( $_REQUEST['nonce'] ) || ! wp_verify_nonce( sanitize_key( $_REQUEST['nonce'] ), 'dokan_admin' ) ) {
+            $permission = false;
+        }
+
+        if ( ! current_user_can( 'manage_woocommerce' ) && $permission ) {
+            $permission = false;
+        }
+
+        return $permission;
+    }
+
+    /**
+     * Retrieves the query params for the dummy data collection.
+     *
+     * @since DOKAN_SINCE
+     *
+     * @return array Collection parameters.
+     */
+    public function get_collection_params() {
```

Comment:
there  is no need for this function, you can directly call `$this->get_endpoint_args_for_item_schema( WP_REST_Server::CREATABLE ),`.  Also add sanitize_callback directly to schema.